core/src/main/java/bisq/core/btc/Balances.java

```diff
@@ -80,18 +80,19 @@ public Balances(TradeManager tradeManager,
     }
 
     public void onAllServicesInitialized() {
-        openOfferManager.getObservableList().addListener((ListChangeListener<OpenOffer>) c -> updateBalance());
-        tradeManager.getObservableList().addListener((ListChangeListener<Trade>) change -> updateBalance());
-        refundManager.getDisputesAsObservableList().addListener((ListChangeListener<Dispute>) c -> updateBalance());
-        xmrWalletService.getWallet().addListener(new MoneroWalletListener() {
-          @Override public void onBalancesChanged(BigInteger newBalance, BigInteger newUnlockedBalance) { updateBalance(); }
-          @Override public void onOutputReceived(MoneroOutputWallet output) { updateBalance(); }
```

Comment:
Assume outputRecieved and outputSpent listeners are not necessary?

Reply:
Right. XmrWalletService notifies listeners of balance changes, and that is sufficient for this class to update itself.